easybuild/scripts/findPythonDeps.py

```diff
@@ -0,0 +1,169 @@
+        if not matching_entries:
+            raise RuntimeError("Found no installed package for '%s' in %s" % (pkg, dep_tree))
+        if len(matching_entries) > 1:
+            raise RuntimeError("Found multiple installed packages for '%s' in %s" % (pkg, dep_tree))
+        entry = matching_entries[0]
+        res.append((entry['package']['package_name'], entry['package']['installed_version']))
+        deps = (dep['package_name'] for dep in entry['dependencies'])
+        res.extend(find_deps(deps, dep_tree))
+    return res
+
+
+def print_deps(package, verbose):
+    if verbose:
+        print('Getting dep tree of ' + package)
+    dep_tree = get_dep_tree(package, verbose)
+    if verbose:
+        print('Extracting dependencies of ' + package)
+    deps = find_deps([extract_pkg_name(package)], dep_tree)
+
+    installed_modules = {mod.project_name for mod in pkg_resources.working_set}
+    if verbose:
+        print("Installed modules: %s" % installed_modules)
+
+    # iterate over deps in reverse order, get rid of duplicates along the way
+    # also filter out Python packages that are already installed in current environment
+    res = []
+    handled = set()
+    for dep in reversed(deps):
+        if dep not in handled:
+            handled.add(dep)
+            if dep[0] in installed_modules:
+                if verbose:
+                    print("Skipping installed module '%s'" % dep[0])
+            else:
+                res.append(dep)
+
+    print("List of dependencies in (likely) install order:")
+    pprint(res, indent=4)
+    print("Sorted list of dependencies:")
+    pprint(sorted(res), indent=4)
+
+
+parser = argparse.ArgumentParser(
+    description='Find dependencies of Python packages by installing it in a temporary virtualenv. ',
+    epilog=' && '.join(['Example usage with EasyBuild: '
+                        'eb TensorFlow-2.3.4.eb --dump-env',
+                        'source TensorFlow-2.3.4.env',
+                        sys.argv[0] + ' tensorflow==2.3.4'])
+)
+parser.add_argument('package', metavar='python-pkg-spec',
+                    help='Python package spec, e.g. tensorflow==2.3.4')
+parser.add_argument('--ec', metavar='easyconfig', help='EasyConfig to use as the build environment')
+parser.add_argument('--verbose', help='Verbose output', action='store_true')
+args = parser.parse_args()
+
+if args.ec:
+    if not can_run('eb', '--version'):
+        print('EasyBuild not found or executable. Make sure it is in your $PATH when using --ec!')
+        sys.exit(1)
+    with temporary_directory() as tmp_dir:
+        old_dir = os.getcwd()
+        os.chdir(tmp_dir)
+        if args.verbose:
+            print('Running EasyBuild to get build environment')
+        run_cmd(['eb', args.ec, '--dump-env', '--force'], action_desc='Dump build environment')
+        os.chdir(old_dir)
+
+        cmd = 'source %s/*.env && %s "%s"' % (tmp_dir, sys.argv[0], args.package)
```

Comment:
This line also assumes the script is executable, which it wasn't in my case when I was trying this out.

Reply:
Changed